Xamarin.Forms.Core/Visuals/VisualTypeConverter.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Xamarin.Forms
+{
+	[Xaml.TypeConversion(typeof(IVisual))]
+	public class VisualTypeConverter : TypeConverter
+	{
+		static Dictionary<string, IVisual> _visualTypeMappings;
+		static Dictionary<string, IVisual> InitMappings()
+		{
+			var mappings = new Dictionary<string, IVisual>(StringComparer.OrdinalIgnoreCase);
+			_visualTypeMappings = mappings;
+			return mappings;
+		}
+
+		public static void Register(string visualKey, IVisual visual)
```

Comment:
The idea here is allowing people get around naming collisions if they need to

Reply:
we don't want registration by code. That prevents Xaml Compilation. If registration is required (i.e. you can't find all the Visuals through assembly reflection), consider registering using assembly attributes.